WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/details/ThreatDetailsViewModel.kt

```diff
@@ -1,44 +1,147 @@
 import org.wordpress.android.ui.jetpack.scan.details.ThreatDetailsViewModel.UiState.Content
 import org.wordpress.android.ui.jetpack.scan.details.usecases.GetThreatModelUseCase
+import org.wordpress.android.ui.jetpack.scan.details.usecases.IgnoreThreatUseCase
+import org.wordpress.android.ui.jetpack.scan.details.usecases.IgnoreThreatUseCase.IgnoreThreatState
+import org.wordpress.android.ui.mysite.SelectedSiteRepository
+import org.wordpress.android.ui.pages.SnackbarMessageHolder
+import org.wordpress.android.ui.utils.HtmlMessageUtils
+import org.wordpress.android.ui.utils.UiString
+import org.wordpress.android.ui.utils.UiString.UiStringRes
+import org.wordpress.android.ui.utils.UiString.UiStringText
+import org.wordpress.android.viewmodel.Event
+import org.wordpress.android.viewmodel.ResourceProvider
 import javax.inject.Inject
+import javax.inject.Named
+
+private const val DELAY_MILLIS = 2000L
 
 class ThreatDetailsViewModel @Inject constructor(
     private val getThreatModelUseCase: GetThreatModelUseCase,
-    private val builder: ThreatDetailsListItemsBuilder
+    private val ignoreThreatUseCase: IgnoreThreatUseCase,
+    private val selectedSiteRepository: SelectedSiteRepository,
+    private val builder: ThreatDetailsListItemsBuilder,
+    private val htmlMessageUtils: HtmlMessageUtils,
+    private val resourceProvider: ResourceProvider,
+    @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher
 ) : ViewModel() {
+    private lateinit var site: SiteModel
     private var isStarted = false
     private var threatId: Long = 0
 
     private val _uiState: MutableLiveData<UiState> = MutableLiveData()
     val uiState: LiveData<UiState> = _uiState
 
+    private val _snackbarEvents = MediatorLiveData<Event<SnackbarMessageHolder>>()
+    val snackbarEvents: LiveData<Event<SnackbarMessageHolder>> = _snackbarEvents
+
+    private val _navigationEvents = MediatorLiveData<Event<ThreatDetailsNavigationEvents>>()
+    val navigationEvents: LiveData<Event<ThreatDetailsNavigationEvents>> = _navigationEvents
+
     fun start(threatId: Long) {
         if (isStarted) {
             return
         }
         isStarted = true
         this.threatId = threatId
+        site = requireNotNull(selectedSiteRepository.getSelectedSite())
         getData()
     }
 
     private fun getData() {
         viewModelScope.launch {
             val model = getThreatModelUseCase.get(threatId)
-            model?.let { updateUiState(it) }
+            model?.let { updateUiState(buildContentUiState(it)) }
+        }
+    }
+
+    private fun ignoreThreat() {
+        viewModelScope.launch {
+            disableThreatActionButtons(true)
+            when (ignoreThreatUseCase.ignoreThreat(site.siteId, threatId)) {
+                is IgnoreThreatState.Success -> {
+                    updateSnackbarMessageEvent(UiStringRes(R.string.threat_ignore_success_message))
+
+                    withContext(bgDispatcher) { delay(DELAY_MILLIS) }
```

Comment:
❓ Are we adding this delay here, so the Snackbar message doesn't disappear right after we show it?

⚠️ If that's the case, this won't work as expected with TalkBack - I believe that delay for snackbar messages when TalkBack is on is set to 10s. Have you considered showing the message on the Scan screen instead of the ThreatDetail screen? I think it'd feel a bit better even for users who don't use TalkBack, wdyt?

Reply:
> Are we adding this delay here, so the Snackbar message doesn't disappear right after we show it?

Correct, delay was added with this 👆intention.

> If that's the case, this won't work as expected with TalkBack - I believe that delay for snackbar messages when TalkBack is on is set to 10s. 

TIL. Thanks for sharing it with me 🙏.

> Have you considered showing the message on the Scan screen instead of the ThreatDetail screen? 

Makes sense, I've added a TODO. I'm thinking to refactor it with fix threat action which will re-use fix all threats implementation in the scan state screen.

Let me know if it's fine to handle it a PR separate to this one.

Reply:
Sounds good to me, thanks!

Reply:
Resolved in https://github.com/wordpress-mobile/WordPress-Android/pull/13825